frontend_tests/node_tests/compose_ui.js

```diff
@@ -118,34 +118,65 @@ run_test("smart_insert", ({override}) => {
         });
     }
     override_with_expected_syntax(" :smile: ");
-    compose_ui.smart_insert($textbox, ":smile:");
+    compose_ui.smart_insert_inline($textbox, ":smile:");
 
     override_with_expected_syntax(" :airplane: ");
-    compose_ui.smart_insert($textbox, ":airplane:");
+    compose_ui.smart_insert_inline($textbox, ":airplane:");
 
     $textbox.caret(0);
     override_with_expected_syntax(":octopus: ");
-    compose_ui.smart_insert($textbox, ":octopus:");
+    compose_ui.smart_insert_inline($textbox, ":octopus:");
 
     $textbox.caret($textbox.val().length);
     override_with_expected_syntax(" :heart: ");
-    compose_ui.smart_insert($textbox, ":heart:");
+    compose_ui.smart_insert_inline($textbox, ":heart:");
 
     // Test handling of spaces for ```quote
     $textbox = make_textbox("");
     $textbox.caret(0);
     override_with_expected_syntax("```quote\nquoted message\n```\n");
-    compose_ui.smart_insert($textbox, "```quote\nquoted message\n```\n");
+    compose_ui.smart_insert_inline($textbox, "```quote\nquoted message\n```\n");
 
     $textbox = make_textbox("");
     $textbox.caret(0);
     override_with_expected_syntax("translated: [Quoting…]\n");
-    compose_ui.smart_insert($textbox, "translated: [Quoting…]\n");
+    compose_ui.smart_insert_inline($textbox, "translated: [Quoting…]\n");
 
     $textbox = make_textbox("abc");
     $textbox.caret(3);
     override_with_expected_syntax(" test with space ");
-    compose_ui.smart_insert($textbox, " test with space");
+    compose_ui.smart_insert_inline($textbox, " test with space");
+
+    // Note that we don't have any special logic for strings that are
+    // already surrounded by spaces, since we are usually inserting things
+    // like emojis and file links.
+});
+
+run_test("smart_insert_block", ({override}) => {
+    let $textbox = make_textbox("abc");
+    $textbox.caret(4);
+    function override_with_expected_syntax(expected_syntax) {
+        override(text_field_edit, "insert", (elt, syntax) => {
+            assert.equal(elt, "textarea");
+            assert.equal(syntax, expected_syntax);
+        });
+    }
+
+    // Test handling of spaces for ```quote
+    $textbox = make_textbox("");
+    $textbox.caret(0);
+    override_with_expected_syntax("```quote\nquoted message\n```\n");
+    compose_ui.smart_insert_block($textbox, "```quote\nquoted message\n```\n");
+
+    $textbox = make_textbox("");
+    $textbox.caret(0);
+    override_with_expected_syntax("translated: [Quoting…]\n");
+    compose_ui.smart_insert_block($textbox, "translated: [Quoting…]\n");
+
+    // $textbox = make_textbox("abc");
+    // $textbox.caret(3);
+    // override_with_expected_syntax(" test with space ");
+    // compose_ui.smart_insert_block($textbox, " test with space");
```

Comment:
Looks like you forgot to remove / uncomment these 4 lines of commented code

Reply:
Thanks @N-Shar-ma! I fixed that by removing the commented code, I believe it just applies to the `smart_insert_inline`

While I was at it, I squashed my commits down to 1, but the Read the Docs build test failed, it looks like it's erroring bc it's missing a historical commit. I'm going to try provisioning, rebasing, and committing again but idk if that will fix that error.